[SYCL][Joint Matrix] Resolve ambiguous fabs calls, refactor tests to remove some duplication.

YuriPlyakhin opened the pull request:
Main purpose of this PR is to resolve ambiguous fabs calls. While doing this, I also did some refactoring to reduce amount of duplication.

dkhaldi reviewed (changes requested):
Are the changes in Legacy necessary as these use short instead of bfloat16 to represent bfloat16 type?

YuriPlyakhin commented:
> Are the changes in Legacy necessary as these use short instead of bfloat16 to represent bfloat16 type?

Yes, changes in Legacy folder are also necessary, since tests there are using fabs without namespace as well.
I did not find where do you see usage of short in my changes. make_fp32 which I removed from Legacy tests and replaced with common.hpp seems to be identical.

YuriPlyakhin commented on sycl/test-e2e/Matrix/Legacy/joint_matrix_bfloat16_colmajorA_colmajorB.cpp:
```diff
@@ -15,12 +15,10 @@
 
 // XFAIL: gpu
 
-#include <iostream>
```
resolved

dkhaldi reviewed (approved):
LGTM

YuriPlyakhin commented:
@intel/llvm-gatekeepers , all green, please, merge
